airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/DataTypeTest.java

```diff
@@ -0,0 +1,179 @@
+  private static final String DEFAULT_INSERT_SQL = "INSERT INTO %1$s VALUES (%2$s, %3$s);";
+
+  private final String sourceType;
+  private final Field.JsonSchemaPrimitive airbyteType;
+  private final List<String> values;
+  private final String createTablePatternSQL;
+  private final String insertPatternSQL;
+  private final String fullSourceDataType;
+  private long testNumber;
+
+  DataTypeTest(String sourceType,
+               Field.JsonSchemaPrimitive airbyteType,
+               List<String> values,
+               String createTablePatternSQL,
+               String insertPatternSQL,
+               String fullSourceDataType) {
+    this.sourceType = sourceType;
+    this.airbyteType = airbyteType;
+    this.values = values;
+    this.createTablePatternSQL = createTablePatternSQL;
+    this.insertPatternSQL = insertPatternSQL;
+    this.fullSourceDataType = fullSourceDataType;
+  }
+
+  /**
+   * The builder allows to setup any comprehensive data type test.
+   *
+   * @param sourceType name of the source data type. Duplicates by name will be tested independently
+   *        from each others. Note that this name will be used for connector setup and table creation.
+   *        If source syntax requires more details (E.g. "varchar" type requires length
+   *        "varchar(50)"), you can additionally set custom data type syntax by
+   *        {@link DataTypeTestBuilder#fullSourceDataType(String)} method.
+   * @param airbyteType corresponding Airbyte data type. It requires for proper configuration
+   *        {@link ConfiguredAirbyteStream}
+   * @return builder for setup comprehensive test
+   */
+  public static DataTypeTestBuilder builder(String sourceType, Field.JsonSchemaPrimitive airbyteType) {
+    return new DataTypeTestBuilder(sourceType, airbyteType);
+  }
+
+  public static class DataTypeTestBuilder {
+
+    private final String sourceType;
+    private final Field.JsonSchemaPrimitive airbyteType;
+    private final List<String> values = new ArrayList<>();
+    private String createTablePatternSQL;
+    private String insertPatternSQL;
+    private String fullSourceDataType;
+
+    DataTypeTestBuilder(String sourceType, Field.JsonSchemaPrimitive airbyteType) {
+      this.sourceType = sourceType;
+      this.airbyteType = airbyteType;
+      this.createTablePatternSQL = DEFAULT_CREATE_TABLE_SQL;
+      this.insertPatternSQL = DEFAULT_INSERT_SQL;
+      this.fullSourceDataType = sourceType;
+    }
+
+    /**
+     * Set custom the create table script pattern. Use it if you source uses untypical table creation
+     * sql. Default patter described {@link #DEFAULT_CREATE_TABLE_SQL} Note! The patter should contains
+     * two String place holders for the table name and data type.
+     *
+     * @param createTablePatternSQL creation table sql pattern
+     * @return builder
+     */
+    public DataTypeTestBuilder createTablePatternSQL(String createTablePatternSQL) {
```

Comment:
@DoNotPanicUA due to JCC createTablePatternSQL->createTablePatternSql

Reply:
fixed everywhere